### PR TITLE
Feature/indexer sanitizing

### PR DESCRIPTION
**Pull request by p0psicles**
- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

Sanitized indexer inputs. Int and Floats are now cast to str by default. I'd like to use it by default like this, as the metadata doesn't handle integer fields very well.
* Sanitize tvmaze input.
* Use WebChannel as network, when available.
* Sanitize tmdb input.
* Use highest value runtime from tmdb.
* Removed some piece of old trakt code. It apparently removes a show from trakt watchlist, when that show can't be added because of "any" reason (is located in an catch Exception block. The main reason I removed it, is because it also still has references to tvrage. Leaving this can cause unexpected situations.

### CODE REVIEW

**Comment by p0psicles** [comment]
Should fix: #1719, #1685

File: medusa/show_queue.py [hunk 1]
```diff
@@ -397,30 +408,32 @@ def run(self):
...
-                    data['shows'][0]['ids']['tvdb'] = self.indexer_id
-                else:
-                    data['shows'][0]['ids']['tvrage'] = self.indexer_id
-
-                try:
-                    trakt_api.traktRequest("sync/watchlist/remove", data, method='POST')
-                except TraktException as e:
-                    logger.log(u"Could not remove show '{0}' from watchlist. Error: {1}".format
-                               (title, e), logger.WARNING)
+            # FIXME: This shouldn't be here. I've commented it. If nobody can provide me with reasons to keep it.
```
Comment (p0psicles) [inline medusa/show_queue.py]: @pymedusa/developers Please tell me, what to do with this?

Comment (fernandog) [inline medusa/show_queue.py]: @p0psicles i used that so im going to explain.

I add a show 1) in trakt watchlist, medusa syncs and add show or 2) in medusa and show is added to trakt watchlist
I watch first episode but didn't like. then i remove show. then medusa removes this show from trakt

if you think it shouldn't be there, at least the call to the function that removes should be.

Comment (p0psicles) [inline medusa/show_queue.py]: Okay that functionality is still there with this Pr. This does only try to remove it from trakt,  if an exception occurs. Which is poorly design and poorly implemented.

Comment (fernandog) [inline medusa/show_queue.py]: but you removed entire code after `app.USE_TRAKT:` 

the call to remove is `trakt_api.traktRequest("sync/watchlist/remove", data, method='POST')`

Comment (p0psicles) [inline medusa/show_queue.py]: Yes, and that's all part of the exception. If there is a specific reason to remove shows with an exception, it should be implemented a different way. Not through a 'catch all' exception.

File: medusa/show_queue.py [hunk 2]
```diff
@@ -397,30 +408,32 @@ def run(self):
...
+            #             }
+            #         ]
+            #     }
+            #     if trakt_id == 'tvdb_id':
+            #         data['shows'][0]['ids']['tvdb'] = self.indexer_id
+            #     # else:
+            #     #     data['shows'][0]['ids']['tvrage'] = self.indexer_id
+            #
+            #     try:
+            #         trakt_api.traktRequest("sync/watchlist/remove", data, method='POST')
```
Comment (fernandog) [inline medusa/show_queue.py]: you shouldn't remove this as this request that removes show from trakt

**Comment by p0psicles** [comment]
@duramato can you run some diverse tests using this branch? If you say go, well merge.

**Comment by duramato** [comment]
Yes i will, anything of more severity i should focus on?

**Comment by p0psicles** [comment]
Focus on the issues you reported. Also includes the imdb link. Try to add some shows on different indexers. And maybe a show that's airing this week. To test the scedule page.

**Comment by p0psicles** [comment]
ooh maybe also test this one again: https://github.com/pymedusa/Medusa/issues/1686.
As that's working fine for me.

**Comment by duramato** [comment]
the invalid timeformat is still present:
image

Also the tt in the imdb link is fixed but for that show it mapped the totally wrong imdb http://www.imdb.com/title/tt0829754/

**Comment by duramato** [comment]
https://github.com/pymedusa/Medusa/issues/1719 is fixed

**Comment by p0psicles** [comment]
My guess is that the show does not have Imdb id mapped. Then it uses the name to attempt to get Imdb id. And there I t gets the wrong one. Can't really do anything about that. Except disabling the het Imdb I'd by name functionality.

**Comment by duramato** [comment]
For https://github.com/pymedusa/Medusa/issues/1685, it shows me a notification i can't add becouse of no seasons but also an error for the exception
image

**Comment by p0psicles** [comment]
Aah yes.  Will need to catch the exception also there.

**Comment by p0psicles** [comment]
Should fix #1729

**Comment by p0psicles** [comment]
Can you test the proper search?

**Comment by fernandog** [comment]
Approved

**Pull request merged**